ResearchKit/Common/ORKWeightPicker.m

```diff
@@ -322,10 +322,11 @@ - (NSArray *)kilogramValues {
     _canonicalMinimumValue = minimumValue;
     _canonicalMaximumValue = maximumValue;
     
-    for (double i = minimumValue; i <= maximumValue; i++) {
+    for (NSUInteger i = minimumValue; i <= maximumValue; i++) {
         [mutableWholeValues addObject:@(i)];
```

Comment:
since minimumValue is a double, I think it's risky to declare "i" as a NSUInteger...

Reply:
Iteration on float values is flagged by the compiler under security IIRC, it needs to be converted to a safer scalar format.

Reply:
Yep: CLANG_ANALYZER_SECURITY_FLOATLOOPCOUNTER. Solid discussion here: https://wiki.sei.cmu.edu/confluence/display/c/FLP30-C.+Do+not+use+floating-point+variables+as+loop+counters

Reply:
Ah thanks for sharing that article!